.gitlab-ci.yml

```diff
@@ -61,20 +63,43 @@ before_script:
       - ${YARN_CACHE_FOLDER}
     policy: pull-push
 
+# ==================== rules ==================== #
+.preview_rules: &preview_rules
+  rules:
+    - if: $CI_COMMIT_BRANCH != $CI_DEFAULT_BRANCH && $CI_COMMIT_BRANCH =~ $PREVIEW_PATTERN
+      when: on_success
+
+.preview_manual_rules: &preview_manual_rules
+  rules:
+    - if: $CI_COMMIT_BRANCH != $CI_DEFAULT_BRANCH && $CI_COMMIT_BRANCH =~ $PREVIEW_PATTERN
+      when: manual
+
+.live_rules: &live_rules
+  rules:
+    - if: $CI_COMMIT_REF_NAME == "master"
+      when: on_success
+
+.live_schedule_rules: &live_schedule_rules
+  rules:
+    - if: $CI_COMMIT_REF_NAME == "master" && $CI_PIPELINE_SOURCE == "schedule"
+      when: on_success
 # ================== templates ================== #
 .base_template: &base_template
   image: public.ecr.aws/x2b9z2t7/webops/site-build:latest
   tags:
     - "runner:main"
-  except:
-    - tags
-  only:
-    - branches
+  rules:
+    - if: $CI_PIPELINE_SOURCE == "tags"
+      when: never
+    - if: $CI_COMMIT_BRANCH != null # Variable doesnt exists on merge request pipelines or tag pipelines.
```

Comment:
can this just be `if: $CI_COMMIT_BRANCH` instead of `if: $CI_COMMIT_BRANCH != null`?

Reply:
@StefonSimmons we probably could, I would need to do some digging but a lot of this is being purposefully explicit for readability for people who may not be fully familiar with how CI setups work. 

We also are doing something similar on the corp side for a rollback variable, makes the code more readable without the need for comments.